src/api/index.ts

```diff
@@ -585,6 +585,70 @@ claimSellerFundsFromSeveralAuctions.sendTransaction = async (
   return DutchX.claimTokensFromSeveralAuctionsAsSeller.sendTransaction(sellArr, buyArr, finalIndices, userAccount)
 }
 
+export const claimAndWithdrawSellerFundsFromSeveralAuctions: ClaimSellerFundsFromSeveralAuctions = async (
+  sell: DefaultTokenObject,
+  buy: DefaultTokenObject,
+  userAccount?: Account,
+  indices: number = 0,
+) => {
+  const { DutchX } = await dxAPI()
+  userAccount = await fillDefaultAccount(userAccount)
+
+  const claimableIndices = (
+    await DutchX.getIndicesWithClaimableTokensForSellers({ sell, buy }, userAccount, indices)
+  )[0].map(i => i.toNumber())
+
+  if (claimableIndices.length === 0) return
+
+  // getIndicesWithClaimableTokensForSellers returns auctions with sellBalance > 0
+  // this means currentAucIndx may have > 0 sellBalance BUT NOT HAVE CLEARED (only last Index though)
+  const lastIndexCleared = (
+    await DutchX.getClosingPrice({ sell, buy }, claimableIndices[claimableIndices.length - 1])
+  )[0].gt(0)
+
+  const length = lastIndexCleared ? claimableIndices.length : claimableIndices.length - 1
+
+  const finalIndices = claimableIndices.slice(0, length)
+  const sellArr = Array.from({ length }, () => sell.address)
+  const buyArr = Array.from({ length }, () => buy.address)
+
+  console.log('Params = ', sellArr, buyArr, finalIndices, userAccount)
+  return DutchX.claimAndWithdrawTokensFromSeveralAuctionsAsSeller(sellArr, buyArr, finalIndices, userAccount)
+}
+
+claimAndWithdrawSellerFundsFromSeveralAuctions.sendTransaction = async (
+  sell: DefaultTokenObject,
```

Comment:
a lot of duplication between this function and the one above

Reply:
lol cause it's the `sendTransaction` version, this is a joke right? or am i missing sth?

Reply:
I mean L530-549, L561-582, L694-613 and L625-646 are the same.
Would have been a nice joke though :smile: